src/Native/LdaNative/model_block.cpp

```diff
@@ -324,14 +337,20 @@ namespace lda
             alias_offset += alias_row_size;
         }
 
-        mem_block_size_ = dict_[num_vocabs_ - 1].end_offset_;
+        uint64_t size = dict_[num_vocabs_ - 1].end_offset_;
+        if (size > numeric_limits<size_t>::max())
+            throw bad_alloc();
+        mem_block_size_ = static_cast<size_t>(size);
         mem_block_ = new int32_t[mem_block_size_]();                // NOTE: force to initialize the values to be zero
 
-        alias_mem_block_size_ = dict_[num_vocabs_ - 1].alias_end_offset_;
+        size = dict_[num_vocabs_ - 1].alias_end_offset_;
+        if (size > numeric_limits<size_t>::max())
+            throw bad_alloc();
+        alias_mem_block_size_ = static_cast<size_t>(size);
         alias_mem_block_ = new int32_t[alias_mem_block_size_]();    //NOTE: force to initialize the values to be zero
 
-        std::cout << "mem_block_size = " << mem_block_size_ * 4 << std::endl;
-        std::cout << "alias_mem_block_size = " << alias_mem_block_size_ * 4 << std::endl;
+        cout << "mem_block_size = " << mem_block_size_ * 4 << endl;
```

Comment:
>4 [](start = 57, length = 1)

same here with sizeof(int32_t)

Reply:
actually instead of using mem_block_size_ * sizeof(uint32_t) -- if you just do sizeof(alias_mem_block_) that will give you the full of the allocated memory.

---
In reply to: [276881438](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276881438) [](ancestors = 276881438)